flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/functions/hive/HiveCountAggFunction.java

```diff
@@ -0,0 +1,116 @@
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.table.types.inference.CallContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.flink.table.expressions.ApiExpressionUtils.unresolvedRef;
+import static org.apache.flink.table.planner.expressions.ExpressionBuilder.ifThenElse;
+import static org.apache.flink.table.planner.expressions.ExpressionBuilder.literal;
+import static org.apache.flink.table.planner.expressions.ExpressionBuilder.plus;
+
+/** built-in hive count aggregate function. */
+public class HiveCountAggFunction extends HiveDeclarativeAggregateFunction {
+
+    private final UnresolvedReferenceExpression count = unresolvedRef("count");
+    private Integer arguments;
+    private boolean count1;
+
+    @Override
+    public int operandCount() {
+        return arguments;
+    }
+
+    @Override
+    public UnresolvedReferenceExpression[] aggBufferAttributes() {
+        return new UnresolvedReferenceExpression[] {count};
+    }
+
+    @Override
+    public DataType[] getAggBufferTypes() {
+        return new DataType[] {DataTypes.BIGINT()};
+    }
+
+    @Override
+    public DataType getResultType() {
+        return DataTypes.BIGINT();
+    }
+
+    @Override
+    public Expression[] initialValuesExpressions() {
+        return new Expression[] {/* count = */ literal(0L, getResultType().notNull())};
+    }
+
+    @Override
+    public Expression[] accumulateExpressions() {
+        // count(*) and count(1) mean that count all elements
+        if (arguments == 0 || count1) {
+            return new Expression[] {/* count = */ plus(count, literal(1L))};
+        }
+
+        // other case need to determine the value of the element
+        List<Expression> operandExpressions = new ArrayList<>();
+        for (int i = 0; i < arguments; i++) {
+            operandExpressions.add(operand(i));
+        }
+        Expression operandExpression =
+                operandExpressions.stream()
+                        .map(ExpressionBuilder::isNull)
+                        .reduce(ExpressionBuilder::or)
+                        .get();
+        return new Expression[] {
+            /* count = */ ifThenElse(operandExpression, count, plus(count, literal(1L)))
+        };
+    }
+
+    @Override
+    public Expression[] retractExpressions() {
+        throw new TableException("Count aggregate function does not support retraction.");
+    }
+
+    @Override
+    public Expression[] mergeExpressions() {
+        return new Expression[] {/* count = */ plus(count, mergeOperand(count))};
+    }
+
+    @Override
+    public Expression getValueExpression() {
+        return count;
+    }
+
+    @Override
+    public void setArguments(CallContext callContext) {
+        if (arguments == null) {
+            arguments = callContext.getArgumentDataTypes().size();
+            if (arguments == 1) {
+                // If the argument is literal 1 indicates use count(1)
```

Comment:
It may be `count(2)`,`count(3)`..., not always 1. May be we can optimize the comment as follows?
```suggestion
                // If the argument is literal, indicates use count(literal)
```
If so, the varaible may also need changes.